src/Riak.php

```diff
@@ -213,10 +226,15 @@ public function getLastRequest()
      * @return $this
      * @throws Exception
      */
-    protected function pickNewNode()
+    public function pickNewNode()
     {
-        // mark current active node as inactive
+        // mark current active node as inactive and increment attempts
         $this->getActiveNode()->setInactive(true);
+        $this->attempts++;
+        $this->inactiveNodes[$this->getActiveNodeIndex()] = $this->getActiveNode();
```

Comment:
Node errors or failures in health check will mark a node as offline. Health check runs periodically to test offline nodes, and if they are found to be online, returns them to the set of running nodes.

This requires the ability to save node state which could be accomplished via shared memory and semaphores in PHP. I may put together a test to see how much a down node affects the PHP client.